windows/hid.c

```diff
@@ -455,6 +479,27 @@ static void hid_internal_get_info(struct hid_device_info* dev)
 	if (cr != CR_SUCCESS)
 		goto end;
 
+	/* Get the hardware ids from devnode */
+	len = 0;
+	cr = CM_Get_DevNode_PropertyW(dev_node, &DEVPKEY_Device_HardwareIds, &property_type, NULL, &len, 0);
+	if (cr == CR_BUFFER_SMALL && property_type == DEVPROP_TYPE_STRING_LIST) {
+		hardware_ids = (wchar_t*)calloc(len, sizeof(BYTE));
+		cr = CM_Get_DevNode_PropertyW(dev_node, &DEVPKEY_Device_HardwareIds, &property_type, (PBYTE)hardware_ids, &len, 0);
+	}
+	if (cr != CR_SUCCESS)
+		goto end;
```

Comment:
are we sure this is the best decision?
in case if we fail to read `HardwareId` why don't we just skip reading the `interface_number` and get to the next property

Reply:
`DEVPKEY_Device_HardwareIds` is mandatory property of any device node in Windows device tree. I think this check should be there just for consistency.

Reply:
ok, at least it is consistent with the other parts in the same function